mhkit/river/io/d3d.py

```diff
@@ -0,0 +1,408 @@
+        v_all=np.append(v_all, layer_data.v)
+    
+    known_points = np.array([ [x, y, z, v] for x, y, z, v in zip(x_all, y_all, z_all, v_all)])
+    
+    all_data= pd.DataFrame(known_points, columns=['x','y','z',f'{variable}'])
+    
+    return all_data
+
+
+def unorm(x, y ,z):
+    '''
+    Calculates the root mean squared value given three arrays. 
+
+    Parameters
+    ----------
+    x: array 
+        One input for the root mean squared calculation.(eq. x velocity) 
+    y: array
+        One input for the root mean squared calculation.(eq. y velocity) 
+    z: array
+        One input for the root mean squared calculation.(eq. z velocity) 
+
+    Returns
+    -------
+    unorm : array 
+        root mean squared output 
+    '''
+
+    assert isinstance(x,(np.ndarray, np.float64, pd.Series)), 'x must be an array'
+    assert isinstance(y,(np.ndarray, np.float64, pd.Series)), 'y must be an array'
+    assert isinstance(z,(np.ndarray, np.float64, pd.Series)), 'z must be an array'
+    
+    if len(x) == len(y) & len (y) ==len (z) :
+        xyz = np.array([x,y,z]) 
+        unorm = np.linalg.norm(xyz, axis= 0)
+    else:
+        raise Exception ('lengths of arrays do not mathch')
+    return unorm
+
+
+
+
+def turbulent_intensity(data, points='cells', time_index= -1, intermediate_values = False ):
+
+    '''
+    Calculated the turbulent intesity for a given data set for the specified points.  
+    Assumes variable names: turkin1, ucx, ucy and ucz.
+
+    Parameters
+    ----------
+    data : netcdf4 object 
+        After running a Delft3D model the output netcdf file can be read into this code. 
+    points : string, DataFrame  
+        Point to interpoate data onto. 
+          'cells' : interpolates all data into velocity coordinat system (Default)
+          'faces': interpolates all dada into TKE coordinate system 
+          DataFrame of x, y, and z corrdinates: Interpolates data onto user povided points 
+    time_step : float 
+        A positive interger to pull the time step from the dataset. Defalt is late time step -1.  
+    intermediate_values: boolean
+        A true or fase boolean that if true will return ucx, uxy, uxz,and turkine1 vaues in Dataframe 
+        if fause will only return x,y,z, and turbulent intesity values. Faules is the default value      
+        
+    Returns
+    -------
+    TI_data : Dataframe
+        If intermediate_values is true all values are output 
+        if intermediate_values is equal to fale only turbulent _intesity and x, y, and z varibles are output 
+            turbulen_intesity: turbulent kinetic energy divided by the root mean squared velocity
+            turkin1: turbulent kinetic energy 
+            ucx: velocity in the x direction 
+            ucy: velocity in the y direction 
+            ucx: velocity in the z direction 
+            x: position in the x direstion 
+            y: position in the y direction 
+            z: position in the z direction 
```

Comment:
we can't use the colon here because of the way numpy builds the docstrings. Let us replace all these colons with a dash